spec/test_support/helper_ruby/aws_helpers.rb

```diff
@@ -0,0 +1,33 @@
+module AwsHelpers
+  # To test our orphan checking code,
+  # we need to be able to feed the
+  # checker an array of fake s3 paths.
+  #
+  # This helper allows you to do something like:
+  #
+  #     paths = ['fake/path/1.jpg', 'fake/path/2.jpg']
+  #     allow(orphan_checker.s3_iterator).to receive(:s3_client).
+  #       and_return(AwsHelpers::MockAwsClient.new(paths: paths).client)
+  #
+  class MockS3Client
```

Comment:
This isn't *that* complicated code -- the heavy living it done by AWS's built in `stub_responses` feature. And we're only using it in one file at present... I would have just done this inline where needed instead of extracting into a class.   

I think the downside is that when reading tests you have to go to another file to see what's really going on, *and* that this class gives the appearance it can do something other than provide a stubbed `list_objects_v2`. 

But if you think it's worth it, that's cool we can leave it extracted -- but I wonder about being clear in comments that ALL this mocks at present is list_objects_v2 call?  Maybe even changing the name of the class to be clear that's kind of all it does?

Reply:
It is used in two other files; will refactor in a separate PR.

Reply:
I'm confident this code belongs in its own helper class. When it was inline (aside from being very un-DRY and easy to get wrong) it just served to distract from the main focus of the tests we were using it in. Now that it's out everything just looks more organized and sane.

Reply:
Good call on the comment; will be more explicit.